components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ai/LoginFlowAIManagerImpl.java

```diff
@@ -0,0 +1,132 @@
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.ai.service.mgt.exceptions.AIClientException;
+import org.wso2.carbon.ai.service.mgt.exceptions.AIServerException;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.ai.service.mgt.util.AIHttpClientUtil.executeRequest;
+import static org.wso2.carbon.identity.application.mgt.ai.constant.LoginFlowAIConstants.AUTHENTICATORS_PROPERTY;
+import static org.wso2.carbon.identity.application.mgt.ai.constant.LoginFlowAIConstants.ErrorMessages.SERVER_ERROR_WHILE_CONNECTING_TO_LOGINFLOW_AI_SERVICE;
+import static org.wso2.carbon.identity.application.mgt.ai.constant.LoginFlowAIConstants.OPERATION_ID_PROPERTY;
+import static org.wso2.carbon.identity.application.mgt.ai.constant.LoginFlowAIConstants.USER_CLAIM_PROPERTY;
+import static org.wso2.carbon.identity.application.mgt.ai.constant.LoginFlowAIConstants.USER_QUERY_PROPERTY;
+import static org.wso2.carbon.registry.core.RegistryConstants.PATH_SEPARATOR;
+
+/**
+ * Implementation of the LoginFlowAIManager interface to communicate with the LoginFlowAI service.
+ */
+public class LoginFlowAIManagerImpl implements LoginFlowAIManager {
+
+    private static final String LOGINFLOW_AI_ENDPOINT = IdentityUtil.getProperty(
+            "AIServices.LoginFlowAI.LoginFlowAIEndpoint");
+    private static final String LOGINFLOW_AI_GENERATE_PATH = "/api/server/v1/applications/loginflow/generate";
+    private static final String LOGINFLOW_AI_STATUS_PATH = "/api/server/v1/applications/loginflow/status";
+    private static final String LOGINFLOW_AI_RESULT_PATH = "/api/server/v1/applications/loginflow/result";
+
+    private static final Log LOG = LogFactory.getLog(LoginFlowAIManagerImpl.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * Generates an authentication sequence using the LoginFlow AI service.
+     *
+     * @param userQuery               The user query. This is a string that contain the requested authentication
+     *                                flow by the user.
+     * @param userClaims              The user claims. This is a JSON array that contains the user claims available
+     *                                for that organization.
+     * @param availableAuthenticators The available authenticators of the organization.
+     * @return Operation ID of the generated authentication sequence.
+     * @throws AIServerException When an error occurs while connecting to the LoginFlow AI service.
+     * @throws AIClientException When an error occurs while generating the authentication sequence.
+     */
+    @Override
+    public String generateAuthenticationSequence(String userQuery, JSONArray userClaims,
```

Comment:
lets rename userClaims to mean the what we are actually getting (the claim metadata)